### ISSUE TITLE
False positive for SC2120 and SC2119

### ISSUE THREAD

**Issue by brother** [koalaman/shellcheck#235]
``` shell
#!/bin/bash

options[0]="first"
options[1]="second"
options[2]="another one"

select_option () {
       while :; do
               echo
               echo "Select one of these:"
               for (( i=0 ; i < ${#options[*]}; i++ )); do
                       echo "$((i+1))) ${options[i]}"
               done | column
               echo
               read -p "Please specify the option to use: (a to abort) "
               if [[ $REPLY == [aA] ]]; then
                       echo -e "\nAborted\n" ; return 1
               elif [[ $REPLY =~ ^[0-9]+$ ]]; then
                       if (( REPLY > ${#options[*]} || REPLY == 0)); then
                               echo "Invalid option selection"
                       else
                               SELECTION=${options[$REPLY-1]%%:*}
                               return
                       fi
               else
                       echo "Unknown input"
               fi
       done
}

if select_option; then
    echo "You selected '$SELECTION'"
else
    echo "You skipped the selection"
fi
```

This will render the following errors:

```
brother /tmp/shellcheck$ shellcheck test.sh 

In test.sh line 7:
select_option () {
^-- SC2120: select_option references arguments, but none are ever passed.


In test.sh line 31:
if select_option; then
   ^-- SC2119: Use select_option "$@" if function's $1 should mean script's $1.
```

Might very well be related to the change in #129. Using 39b88bb (and shellcheck.net =)).
